snake.go

```diff
@@ -84,9 +84,8 @@ func ToScreamingDelimited(s string, delimiter uint8, ignore string, screaming bo
 			vIsNum := v >= '0' && v <= '9'
 			nextIsCap := next >= 'A' && next <= 'Z'
 			nextIsLow := next >= 'a' && next <= 'z'
-			nextIsNum := next >= '0' && next <= '9'
 			// add underscore if next letter case type is changed
-			if (vIsCap && (nextIsLow || nextIsNum)) || (vIsLow && (nextIsCap || nextIsNum)) || (vIsNum && (nextIsCap || nextIsLow)) {
+			if (vIsCap && nextIsLow || vIsLow && nextIsCap || vIsNum && nextIsCap) {
```

Comment:
In your new logic, for the first two terms you are dropping the `|| nextIsNum` part, but in the last term, you are dropping the `|| nextIsLow` part. Was this intentional?

Reply:
Yes, the idea was that a number followed by a capital letter is still treated as a change of case, but a letter followed by a number is not.